engine/moduleManager/modules/input/InputModule.hpp

```diff
@@ -0,0 +1,32 @@
+//
+// Created by 11896 on 26/11/2024.
+//
+
+#ifndef BLOCKYENGINE_ENGINE_MODULEMANAGER_MODULES_INPUT_INPUTMODULE_HPP_
+#define BLOCKYENGINE_ENGINE_MODULEMANAGER_MODULES_INPUT_INPUTMODULE_HPP_
+
+#include <functional>
+#include "KeyEvent.hpp"
+#include "MouseEvent.hpp"
+#include "SDL.h"
+#include "BlockyEngine.hpp"
+#include "MouseInput.hpp"
+
+class InputModule {
+	public:
+		InputModule() = default;
+
+		void PollEvents();
+		void AddKeyListener(const std::function<void(KeyEvent)>& listener);
+		void RemoveKeyListener(const std::function<void(KeyEvent)>& listener);
+		void AddMouseListener(const std::function<void(MouseEvent)>& listener);
+		void RemoveMouseListener(const std::function<void(MouseEvent)>& listener);
+		static std::pair<int, int> GetCursorPosition() ;
+
+	private:
+		std::vector<std::function<void(KeyEvent)>> _keyListeners;
```

Comment:
Ik zou dit opslaan als listeners op bepaalde keys. dus in een std::map<KeyInput, std::vector<std::function<void(KeyState)>>>. Hierbij hebben de callbacks dus alleen nodig of het UP of DOWN is. Dit zou veel if-statements voorkomen in de callbacks gezien je dan gwn een callback op een specifieke knop maakt. Ook te overwegen bij mouse